src/goto-programs/goto_convert.cpp

```diff
@@ -1184,6 +1184,24 @@ void goto_convertt::convert_switch(
   goto_programt sideeffects;
   clean_expr(argument, sideeffects, mode);
 
+  // make sure evaluating the value multiple times (as the below chain-of-ifs
+  // does) doesn't have unintended effects
+  if(needs_cleaning(argument))
```

Comment:
I would change the comment: "unintended effects" sounds a bit like "side effect", which should be removed at this point by `clean_expr`.

I'd be clear that this is about reducing the cost of multiple copies of `argument`. This could well be independent of `needs_cleaning`. E.g., forall/exist "do not need cleaning", but may well be expensive.

Reply:
Makes sense, but I did wonder about concurrency: in that situation, repeat evaluation might actually change the semantics of the program. Not that `needs_cleaning` currently covers this, though...

Reply:
Concurrency is naughty. Repeated evaluation would certainly be something a compiler would be allowed to do. One-time evaluation would only be guaranteed if the variables in the expression are `volatile`.

Reply:
Comment updated to reflect your input.

Reply:
`does) does` -> `does)`

Reply:
Thank you for careful proof reading!